convert_osm/src/lib.rs

```diff
@@ -181,14 +181,17 @@ fn clip_map(map: &mut RawMap, timer: &mut Timer) {
 
     let mut result_areas = Vec::new();
     for orig_area in map.areas.drain(..) {
-        for polygon in map
+        // If clipping fails, giving up on some areas is fine
```

Comment:
A real example in Milwaukee:
![Screenshot from 2022-08-30 09-50-18](https://user-images.githubusercontent.com/1664407/187399469-5822a998-198d-4a92-be67-756646f23fe8.png)
The red polygon disappears with this PR. For some reason, clipping https://www.openstreetmap.org/relation/6041452#map=17/43.05795/-87.87511 to the map's boundary produces an invalid ring somewhere. But as you can see from the "before" screenshot above, the final result used to work fine.

Reply:
Yeah, it'd be great if One Day™ we added some validation rules, and maybe some heuristics for "guessing" at how to best make an existing invalid geometry valid. e.g. something like https://github.com/libgeos/geos/blob/main/src/operation/valid/MakeValid.cpp

FYI there's a related long standing issue open about implementing just the validity checks: https://github.com/georust/geo/issues/123 but it hasn't seen activity in a long time.